repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -997,17 +997,27 @@ private void mapAttributes(AtlasStruct struct, AtlasStructType structType, Atlas
                         Object attrOldValue = null;
                         boolean isArrayOfPrimitiveType = false;
                         boolean isArrayOfEnum = false;
+
+                        boolean isStruct = (TypeCategory.STRUCT == attribute.getDefinedInType().getTypeCategory()
+                                || TypeCategory.STRUCT == attribute.getAttributeType().getTypeCategory());
+
                         if (attribute.getAttributeType().getTypeCategory().equals(ARRAY)) {
```

Comment:
should we add a check here for getDefinedInType with OR operator? (just like isStruct check)